Explain public streams and filters (#449)
dolittle/Runtime: Documentation/Concepts/streams.md

(((2 x event handlers) + filters) x tenants)  + event horizon subscriptions = stream processors
```
{{< /alert >}}

> ## Public Streams
> 
> We differentiate between two different types of event streams; public and private. Public streams are, in addition to what we have explained above, simply streams that are exposed to other microservices. Through the [Event Horizon]({{< ref "event_horizon.md" >}}) other microservices can subscribe to your public streams. Using a [public filter]({{< ref "event_handles_and_filters.md#public-filters" >}}) you can filter out [public events]({{< ref "events.md#public-vs.-private" >}}) to public streams.

Reviewer: ```suggestion
### Public vs. Private Streams
There are two different types of event streams; public and private. Private streams are exposed within their [Tenant]({{< ref "#multi-tenancy" >}}) and public streams are additionally exposed to other microservices. Through the [Event Horizon]({{< ref "event_horizon.md" >}}) other microservices can subscribe to your public streams. Using a [public filter]({{< ref "event_handles_and_filters.md#public-filters" >}}) you can filter out [public events]({{< ref "events.md#public-vs.-private" >}}) to public streams.
```

I suggest we move this section to the top, right before Rules as it is talking about just streams and their properties. Being all the way at the bottom of the after the processors seems a bit too deep into the page.

Also reworded the sentence a bit, maybe better idk.
